### DISCUSSION TITLE
Need help with manual setup of RustDesk server

### DISCUSSION THREAD

**Discussion by Logansfury** [rustdesk/rustdesk-server#327] [Q&A]
Hello,

I am trying to setup my Linux Mint box for remote desktop login by my windows machines. I have installed RustDesk, PM2 and NodeJS according to all the instructions and linked instructions on this page:

https://rustdesk.com/docs/en/self-host/rustdesk-server-oss/install/#set-up-your-own-server-instance-manually

but something seems to be going wrong when I use the start commands:

logansfury@OptiPlex-5040:~$ pm2 start hbbr
[PM2] Starting /usr/bin/hbbr in fork_mode (1 instance)
[PM2] Done.
┌────┬────────────────────┬──────────┬──────┬───────────┬──────────┬──────────┐
│ id │ name               │ mode     │ ↺    │ status    │ cpu      │ memory   │
├────┼────────────────────┼──────────┼──────┼───────────┼──────────┼──────────┤
│ 1  │ hbbr               │ fork     │ 0    │ stopped   │ 0%       │ 0b       │
│ 0  │ hbbs               │ fork     │ 15   │ errored   │ 0%       │ 0b       │
└────┴────────────────────┴──────────┴──────┴───────────┴──────────┴──────────┘
logansfury@OptiPlex-5040:~$ 

Setting this up successfully means that I can edit the random number ID logons to alphabetic usernames, and remote in from outside my home intranet correct?

Can anyone explain why this is stopped and errored instead of running?

Thank you for reading,

Logan

**Comment by danielmarschall** [comment]
Setting up hbbr and hbbs mean that you have your own network and are independent from the public servers out there. Not sure about alphabetic usernames, I never used something like that.

What happens when you run `./hbbs` and `./hbbr` manually in the console? Any error messages?

**Comment by Logansfury** [comment]
Hello Daniel,

Thank you very much for taking the time to respond. I cannot launch these services without using PM2:

logansfury@OptiPlex-5040:~$ ./hbbs
bash: ./hbbs: No such file or directory
logansfury@OptiPlex-5040:~$ ./hbbr
bash: ./hbbr: No such file or directory
logansfury@OptiPlex-5040:~$ 

with the command $ pm2 start hbbs:

logansfury@OptiPlex-5040:~$ pm2 start hbbs
[PM2] Applying action restartProcessId on app [hbbs](ids: [ 0 ])
[PM2] [hbbs](0) ✓
[PM2] Process successfully started
┌────┬────────────────────┬──────────┬──────┬───────────┬──────────┬──────────┐
│ id │ name               │ mode     │ ↺    │ status    │ cpu      │ memory   │
├────┼────────────────────┼──────────┼──────┼───────────┼──────────┼──────────┤
│ 1  │ hbbr               │ fork     │ 15   │ errored   │ 0%       │ 0b       │
│ 0  │ hbbs               │ fork     │ 15   │ online    │ 0%       │ 0b       │
└────┴────────────────────┴──────────┴──────┴───────────┴──────────┴──────────┘
logansfury@OptiPlex-5040:~$ 

pm2 start hbbr:

logansfury@OptiPlex-5040:~$ pm2 start hbbr
[PM2] Applying action restartProcessId on app [hbbr](ids: [ 1 ])
[PM2] [hbbr](1) ✓
[PM2] Process successfully started
┌────┬────────────────────┬──────────┬──────┬───────────┬──────────┬──────────┐
│ id │ name               │ mode     │ ↺    │ status    │ cpu      │ memory   │
├────┼────────────────────┼──────────┼──────┼───────────┼──────────┼──────────┤
│ 1  │ hbbr               │ fork     │ 15   │ stopped   │ 0%       │ 0b       │
│ 0  │ hbbs               │ fork     │ 30   │ errored   │ 0%       │ 0b       │
└────┴────────────────────┴──────────┴──────┴───────────┴──────────┴──────────┘
logansfury@OptiPlex-5040:~$ 


Would you know why its showing stopped and errored?

**Reply by danielmarschall** [reply]
Can you check where the binary files hbbs and hbbr are stored? You might find that out by running "whereis hbbr" or "locate hbbr"

**Comment by Logansfury** [comment]
> Can you check where the binary files hbbs and hbbr are stored? You might find that out by running "whereis hbbr" or "locate hbbr"

Here is the output:

logansfury@OptiPlex-5040:~$ whereis hbbr
hbbr: /usr/bin/hbbr
logansfury@OptiPlex-5040:~$ whereis hbbs
hbbs: /usr/bin/hbbs
logansfury@OptiPlex-5040:~$

**Reply by danielmarschall** [reply]
Ok, then please run `/usr/bin/hbbs` and/or `/usr/bin/hbbr`, and see if there are errors. You can also run both if you have two console windows. In case something fails, you can see the error message.

**Reply by Logansfury** [reply]
It looks like despite errors of "stopped" and "errored" in the terminal the RustDesk setup is working. I put RustDesk on my win7 laptop went down the street to my local Barnes and Noble Bookstore/Internet Cafe logged on their internet and was able to connect right into my Linux box back at home :)

**Comment by Logansfury** [comment]
logansfury@OptiPlex-5040:~$ /usr/bin/hbbs
[2023-12-02 10:57:21.398309 -08:00] INFO [src/common.rs:121] Private key comes from id_*******
[2023-12-02 10:57:21.398328 -08:00] INFO [src/peer.rs:84] DB_URL=./db_v2.sqlite3
[2023-12-02 10:57:21.399385 -08:00] INFO [src/rendezvous_server.rs:98] serial=0
[2023-12-02 10:57:21.399393 -08:00] INFO [src/common.rs:46] rendezvous-servers=[]
[2023-12-02 10:57:21.399395 -08:00] INFO [src/rendezvous_server.rs:100] Listening on tcp/udp :21116
[2023-12-02 10:57:21.399397 -08:00] INFO [src/rendezvous_server.rs:101] Listening on tcp :21115, extra port for NAT test
[2023-12-02 10:57:21.399399 -08:00] INFO [src/rendezvous_server.rs:102] Listening on websocket :21118
[2023-12-02 10:57:21.399419 -08:00] INFO [libs/hbb_common/src/udp.rs:35] Receive buf size of udp [::]:21116: Ok(212992)
[2023-12-02 10:57:21.399444 -08:00] INFO [libs/hbb_common/src/udp.rs:35] Receive buf size of udp 0.0.0.0:21116: Ok(212992)
Error: Address in use (os error 98)
logansfury@OptiPlex-5040:~$

**Comment by Logansfury** [comment]
Why are all my pastes showing with stroke-thru font?

**Reply by danielmarschall** [reply]
because of the tilde symbols. You should format them as code or quote.

"Error: Address in use (os error 98)" probably means that the port is already used by another process (a different instance of hbbs/hbbr?)

Before running hbbr / hbbr, maybe you should do a "killall hbbs" and "killall hbbr" to make sure any leftover processes are terminated?

**Reply by Logansfury** [reply]
These commands didn't work:

    logansfury@OptiPlex-5040:$ killall hbbs
    hbbs(697): Operation not permitted
    hbbs: no process found
    logansfury@OptiPlex-5040:$ killall hbbr
    hbbr(696): Operation not permitted
    hbbr: no process found
    logansfury@OptiPlex-5040:~$

**Comment by Logansfury** [comment]
It's working!

Despite the "stopped" and "errored" messages in the terminal, I was able to load RustDesk onto my windows 7 laptop, bring the laptop to an internet cafe, and log right into my Linux box and start manipulating things.
